api-inference-community/requirements.txt

```diff
@@ -1,7 +1,7 @@
-starlette==0.14.2
+starlette>=0.14.2
 numpy>=1.18.0
-pydantic==1.8.1
-parameterized==0.8.1
-pillow==8.2.0
+pydantic>=1.8.1
+parameterized>=0.8.1
+pillow>=8.2.0
 huggingface_hub>=0.0.12
-datasets==1.10
+datasets>=1.10
```

Comment:
apart from pillow, are the other losse specifiers required?

Reply:
well this is a library, so everything should be smooth to allow users to get whatever they want, it's in the application layer you want to pin specific versions to get reproducible builds.

Technically to be 100% safe you should always do =>Major,<Major+1 to avoid breaking changes when they are released.

But since we use such a shallow surface I think we're ok not safeguarding us from this.

Reply:
Just to emphasize, the real culprit here was not really Pillow==8.2.0 (only to some extent).

The biggest culprit is `datasets==1.10` because datasets itself restricts `huggingface_hub<0.1` which is now at `0.4`  So This `huggingface_hub` incompatibility forces `pip` to look for earlier versions of packages(not entirely sure about the traversal order). Pip will attempt a bunch of versions before dying on `Pillow` strict requirements, but modifying only `Pillow` doesn't fix the incompatibility issues, it would just send the incompatibility somewhere else. We could have fixed them one by one until fixing `datasets` in the end (it's the first issue reported by pip when doing the install).